pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -185,7 +257,7 @@ func (a *sqlInstanceAdapter) Create(ctx context.Context, u *unstructured.Unstruc
 				// Ref: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance
```

Comment:
One problem with doing more than one thing in a reconciliation is what happens if the first one succeeds but the second fails.  So we might want to extract this into a function, so that we can also call it when handling the adoption case.  But of course, then we need to be careful that we don't delete the root user if the user created it.

(Arguably, this is why it's not a good idea to do things like this in the IaC layer, particularly if the user has not asked us to)

Reply:
Agreed. Though, maybe we don't need to delete the user at all? This was only done to achieve implementation parity with Terraform. Perhaps it would be ok to just.. not do this? Wdyt?

Reply:
I don't think we want to change the behaviour here.  But yeah, I'm sure it will come up again...